src/openstudio_lib/VariablesTabView.cpp

```diff
@@ -139,34 +166,89 @@ void VariableListItem::onOffClicked(bool t_on) {
   }
 }
 
-VariablesList::VariablesList(openstudio::model::Model t_model) : m_model(t_model), m_dirty(true) {
+VariablesList::VariablesList(openstudio::model::Model t_model) : m_model(t_model), m_dirty(true), m_searchActive(false) {
   t_model.getImpl<openstudio::model::detail::Model_Impl>()->addWorkspaceObject.connect<VariablesList, &VariablesList::onAdded>(this);
 
   t_model.getImpl<openstudio::model::detail::Model_Impl>()->removeWorkspaceObject.connect<VariablesList, &VariablesList::onRemoved>(this);
+
+  // TODO: should probably use a GridLayout, this is getting out of control
+
   auto* vbox = new QVBoxLayout();
   vbox->setContentsMargins(10, 10, 10, 10);
   vbox->setSpacing(10);
   setLayout(vbox);
 
-  vbox->addWidget(new QLabel("<b>Possible Output Variables</b>"));
+  auto* outerHBox = new QHBoxLayout();
+
+  auto* allOnOffVBox = new QVBoxLayout();
+  {
+    allOnOffVBox->addWidget(new QLabel("<b>Possible Output Variables</b>"));
+
+    m_allOnBtn = new QPushButton("All On");
+    m_allOnBtn->setFlat(true);
+    m_allOnBtn->setObjectName("StandardGrayButton");
+    m_allOffBtn = new QPushButton("All Off");
+    m_allOffBtn->setFlat(true);
+    m_allOffBtn->setObjectName("StandardGrayButton");
+    allOnOffVBox->addWidget(m_allOnBtn);
+    allOnOffVBox->addWidget(m_allOffBtn);
 
-  auto* innerbox = new QVBoxLayout();
-  m_allOnBtn = new QPushButton("All On");
-  m_allOnBtn->setFlat(true);
-  m_allOnBtn->setObjectName("StandardGrayButton");
-  m_allOffBtn = new QPushButton("All Off");
-  m_allOffBtn->setFlat(true);
-  m_allOffBtn->setObjectName("StandardGrayButton");
-  innerbox->addWidget(m_allOnBtn);
-  innerbox->addWidget(m_allOffBtn);
+    connect(m_allOnBtn, &QPushButton::clicked, this, &VariablesList::allOnClicked);
+    connect(m_allOffBtn, &QPushButton::clicked, this, &VariablesList::allOffClicked);
+  }
+
+  outerHBox->addLayout(allOnOffVBox);
```

Comment:
I think my main comment on this PR would be to remove the separated frame, then make the "All On" and "All Off" buttons apply to the currently visible output variables.  That would make it easier to turn on/off all variables for "Surfaces" for example.  The current implementation of turning on/off all variables in the model is not useful.

Reply:
Here's an idea for the layout, I'd be happy to help on this if you'd like or you can go for it.

![image](https://github.com/openstudiocoalition/OpenStudioApplication/assets/1327850/996b9019-5c5a-4cad-b75e-68078a266d15)

Reply:
Any idea how up to date OpenStudio has been keeping the `ModelObject::outputVariableNames` methods?  I wonder if it would be better to use the RDD instead if it exists?

Reply:
I would reckon some output variable are outdated as changes slipped through the cracks, but we do try to keep em up to date as part of the update to new E+ IO Freeze builds

Reply:
The "All On" / "All Off" only applies to currently visible variables already. You can also see it in action in the GIF above.

If you think your layout is clearer I'm not opposed to it.

Reply:
Moved to https://github.com/openstudiocoalition/OpenStudioApplication/issues/626